### PR TITLE
Gen2 common objdet

### PR DESCRIPTION
**Pull request by jonngai**

### CODE REVIEW

File: include/depthai/pipeline/node/CommonObjDet.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#pragma once
```
Comment (themarpe) [inline include/depthai/pipeline/node/CommonObjDet.hpp:1]: Still used?

File: include/depthai/pipeline/node/DetectionNetwork.hpp [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    std::shared_ptr<Node> clone() override;
+
+    OpenVINO::Version networkOpenvinoVersion;
+
+   protected:
+    DetectionNetwork(const std::shared_ptr<PipelineImpl>& par, int64_t nodeId);
+    dai::DetectionNetworkProperties properties;
+
+   public:
+    Input input{*this, "in", Input::Type::SReceiver, {{DatatypeEnum::Buffer, true}}};
```
Comment (themarpe) [inline include/depthai/pipeline/node/DetectionNetwork.hpp:30]: Should only accept ImgFrame types (as input w&h are required, right?)
```
Input input{*this, "in", Input::Type::SReceiver, {{DatatypeEnum::ImgFrame, false}}};
```

File: include/depthai/pipeline/node/DetectionNetwork.hpp [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    Output out{*this, "out", Output::Type::MSender, {{DatatypeEnum::Buffer, false}}};
```
Comment (themarpe) [inline include/depthai/pipeline/node/DetectionNetwork.hpp:31]: Outputs ImgDetections type:
```
Output out{*this, "out", Output::Type::MSender, {{DatatypeEnum::ImgDetections, false}}};
```

**Review by themarpe — CHANGES_REQUESTED** [review]
Above mentioned comments and please remove `standaloneYOLOExample`. Move to `depthai-experiments`

File: include/depthai/pipeline/node/CommonObjDet.hpp (hunk 1 above)
Comment (jonngai) [reply include/depthai/pipeline/node/CommonObjDet.hpp:1]: Oh, good call, I can remove all the old CommonObjDet stuff.

File: include/depthai/pipeline/node/DetectionNetwork.hpp (hunk 3 above)
Comment (jonngai) [reply include/depthai/pipeline/node/DetectionNetwork.hpp:31]: The SPIOut node is expecting DatatypeEnum::Buffer for input though. Can I leave it as DatatypeEnum::Buffer or is there some other way around this?

File: include/depthai/pipeline/node/DetectionNetwork.hpp (hunk 2 above)
Comment (jonngai) [reply include/depthai/pipeline/node/DetectionNetwork.hpp:30]: Changed.

File: include/depthai/pipeline/node/DetectionNetwork.hpp (hunk 3 above)
Comment (themarpe) [reply include/depthai/pipeline/node/DetectionNetwork.hpp:31]: SPIOut is expecting Buffer or its descendants (think, derived objects / polymorphism), same as XLinkIn.

Therefore they can both accept any messages.

**Comment by themarpe** [comment]
@jonngai One extra request - can you derive `DetectionNetwork` from `NeuralNetwork`?

**Comment by jonngai** [comment]
> @jonngai One extra request - can you derive `DetectionNetwork` from `NeuralNetwork`?

Hey @themarpe , I set that up but still had to override a number of methods since DetectionNetwork has its own inputs, outputs and properties. Let me know if there's a better way of handling this or if this'll do for the time being. Thanks.

File: include/depthai/pipeline/node/DetectionNetwork.hpp (hunk 3 above)
Comment (jonngai) [reply include/depthai/pipeline/node/DetectionNetwork.hpp:28]: Probably a noob question, but when using the inherited getOutputs and getInputs methods, won't they return the base class's input and out?

**Pull request merged**